src/view/widgets/Interactive.ts

```diff
@@ -0,0 +1,304 @@
+
+        return maxlength ? parseInt(maxlength) : undefined;
+    }
+
+    public setMaxLength(maxlength: number): this {
+        return this.setAttribute("maxlength", maxlength);
+    }
+
+    public getPlaceholder(): Option<string> {
+        return this.getAttribute("placeholder");
+    }
+
+    public setPlaceholder(placeholder: string): this {
+        return this.setAttribute("placeholder", placeholder);
+    }
+
+    public getSize(): Option<number> {
+        const size = this.getAttribute("size");
+
+        return size ? parseInt(size) : undefined;
+    }
+
+    public setSize(size: number): this {
+        return this.setAttribute("size", size);
+    }
+
+	public getValue(): Option<string> {
+		return this.getAttribute("value");
+	}
+
+    public setValue(value: string): this {
+        return this.setAttribute("value", value);
+    }
+
+	public override build(): string {
+		return /*html*/ `
+            <vscode-text-field ${this.getFormattedAttributes()}>
+                ${this.label ?? ""}
+                ${this.startIcon ? `<span slot="start" class="codicon codicon-${this.startIcon}"></span>` : ""}
+                ${this.endIcon ? `<span slot="end" class="codicon codicon-${this.endIcon}"></span>` : ""}
+                ${this.buildChildren()}
+            </vscode-text-field>
+        `;
+	}
+}
+
+export class AudioPlayback extends Widget {
+
+    public static readonly RESOURCES = Resources.fromCSS(`
+        audio {
+            transform: scale(60%);
+        }
+
+        audio::-webkit-media-controls-enclosure {
+            background-color: var(--button-secondary-background);
+            border-radius: calc(var(--corner-radius-round) * 1px);
+        }
+
+        .vscode-dark audio {
+            color-scheme: dark;
+        }
+    `);
+
+    protected readonly src: string;
+
+    protected copiedWebviewUri?: Result<Uri>;
+
+	constructor(properties: MergeProperties<{
+        src: string
+    }>) {
+		super(properties);
+
+		this.src = properties.src;
+	}
+
+    public getSrc(): string {
+        return this.src;
+    }
+
+	public override build(): string {
+		return /*html*/ `
+			<div ${this.getFormattedAttributes()}>
+				${this.copiedWebviewUri?.isValue() ? /*html*/ `
+                    <audio controls>
+                        <source src="${this.copiedWebviewUri.unwrap()}">
+                    </audio>
+                ` : (this.copiedWebviewUri ? /*html*/ `
+                    <p>Unable to load audio: ${this.copiedWebviewUri?.unwrapErr()}</p>
+                ` : "")}
+				${this.buildChildren()}
+			</div>
+        `;
+	}
+
+    protected override postInit(): void {
+        this.copiedWebviewUri = this.generateTempCopy(this.src);
+    }
+}
+
+// I know that the redirecting in this sucks,
+// I tried to fix it but someone at VSCode decided to dial the content security policy to 11 so I can't access anything which allows me to catch redirects.
+export class IFrame extends Element {
```

Comment:
Ye it's literally possible to fix iframe navigation. Spent a whole day trying to find a way.